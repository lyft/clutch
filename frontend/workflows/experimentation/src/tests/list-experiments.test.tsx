import React from "react";
import { shallow } from "enzyme";

import ListExperiments from "../list-experiments";

jest.mock("@clutch-sh/core", () => {
  return {
    ...(jest.requireActual("@clutch-sh/core") as any),
    useNavigate: jest.fn(),
  };
});

describe("List Experiments workflow", () => {
  let links;
  let columns;

  beforeAll(() => {
    links = [
      {
        displayName: "button_1",
        path: "/path1",
      },
    ];
    columns = [
      {
        id: "column_1",
        header: "column 1",
      },
      {
        id: "column_2",
        header: "column 2",
      },
    ];
  });

  it("renders correctly", () => {
    const component = shallow(
      <ListExperiments heading="List Experiments" columns={columns} links={links} />
    );
    expect(component.debug()).toMatchSnapshot();
  });
});
