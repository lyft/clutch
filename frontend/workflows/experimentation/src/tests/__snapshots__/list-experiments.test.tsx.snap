// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Experiments workflow renders correctly 1`] = `
"<PageLayout heading=\\"List Experiments\\" error=\\"\\">
  <ButtonGroup buttons={{...}} />
  <ListView columns={{...}} items={[undefined]} onRowSelection={[Function: onRowSelection]} />
</PageLayout>"
`;
